### PR TITLE
Create setup script

### PR DESCRIPTION
**Pull request by cabarnes**
Create a script to help with setting up the necessary dependencies for running the application. Right now the script will only work with Debian-based Linux distributions.

I have also updated the Dockerfile to keep it in sync.

This addresses issue #36.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,13 +3,15 @@ FROM debian:stable
 MAINTAINER Clifton Barnes <clifton.a.barnes@gmail.com>
 
 RUN apt-get update
-RUN apt-get install -y python python-dev python-pip lighttpd
+RUN apt-get install -y python python-dev python-pip python-smbus lighttpd
```
Comment (aninternetof) [inline Dockerfile]: What requires python-smbus?

File: Dockerfile [hunk 2]
```diff
@@ -3,13 +3,15 @@ FROM debian:stable
...
 RUN pip install flask flask_cors
```
Comment (aninternetof) [inline Dockerfile]: We don't need flask_cors anymore (horray!)

File: setup.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	exit 1
+fi
+
+if [ ! -e ${ADAFRUIT_DIR} ]; then
+	echo "Error: Adafruit directory not found."
+	exit 2
+fi
+
+apt-get install -y python python-dev python-pip python-smbus lighttpd
+pip install flask flask_cors
```
Comment (aninternetof) [inline setup.sh]: Don't need flask_cors anymore

File: setup.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	echo "Error: This script must be run as root."
+	exit 1
+fi
+
+if [ ! -e ${ADAFRUIT_DIR} ]; then
+	echo "Error: Adafruit directory not found."
+	exit 2
+fi
+
+apt-get install -y python python-dev python-pip python-smbus lighttpd
```
Comment (aninternetof) [inline setup.sh]: What requires python-smbus?

Comment (aninternetof) [inline setup.sh]: If we're going to install lighttpd for them, maybe we go as far as to provide them with the lighttpd.conf that they'll need (with the mod_proxy set to foward to our port 5000 Flask app)?
We could either edit the default lighttpd.conf with this script, or keep ours in git and copy it over their default one.
Thoughts?

File: Dockerfile (hunk 1 above)
Comment (cabarnes) [inline Dockerfile]: The Adafruit [README](https://github.com/adafruit/Adafruit_Python_GPIO/blob/master/README.md)

File: Dockerfile (hunk 2 above)
Comment (cabarnes) [inline Dockerfile]: I'll add the change to this pull request, but this pull request will then need to be after #44.

File: setup.sh (hunk 4 above)
Comment (cabarnes) [inline setup.sh]: I think including a default one would be better. This script has no idea what version of lighttpd will be installed, so trying to edit the default configuration file that might be slightly different between versions doesn't seem practical.

File: Dockerfile (hunk 1 above)
Comment (aninternetof) [inline Dockerfile]: Hm. I think Adafruit only requires that for using I2C, which we're not using. But we might one day. And if they call it a requirement, we'll let it be a requirement.

File: Dockerfile (hunk 2 above)
Comment (aninternetof) [inline Dockerfile]: Oh, right, I forgot that that change is still in review.

File: setup.sh (hunk 4 above)
Comment (aninternetof) [inline setup.sh]: Cool.
I think the commit of the lighttpd.conf makes sense to come with this pull request. [Here is mine for reference](https://gist.github.com/aninternetof/4432f1428fec367fe6ebd20207ef5766). I guess it won't need the php cgi stuff for much longer, but we can remove it later.

**Comment by wavetylor** [comment]
Looks good!

**Comment by cabarnes** [comment]
I went ahead and included the lighttpd.conf that I already had working. Once we move to nginx we can remove it, but it will at least allow all of this to work.

File: setup.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+pip install flask
+
+pushd ${ADAFRUIT_DIR} > /dev/null
+python setup.py install
+popd > /dev/null
```
Comment (aninternetof) [inline setup.sh]: what do pushd and popd do?

**Pull request merged**